lib/dependencies/inspect-implementation.ts

```diff
@@ -12,8 +12,17 @@ export function getMetaData(
   root: string,
   targetFile: string
 ) {
+
+  // if we're dealing with a Pipfile, lets not assume its in CWD, this
+  // ensures pipenv is always dealing with pipfile we want it to
+  if ( path.basename(targetFile) === 'Pipfile') {
+    var PIPENV_OVERRIDE = { ...process.env, "PIPENV_PIPFILE": targetFile };
+  } else {
+    var PIPENV_OVERRIDE = { ...process.env, "PIPENV_PIPFILE": "" };
+  }
+
   return subProcess
-    .execute(command, [...baseargs, '--version'], { cwd: root })
+    .execute(command, [...baseargs, '--version'], { cwd: root, env: PIPENV_OVERRIDE })
```

Comment:
Bear in mind that explicitly setting `env` means none of the existing `env` gets passed through to the child process. You might need to merge PIPENV_OVERRIDE into the existing env

Reply:
PIPENV_OVERRIDE is set to include all existing process.env when it's declared

Reply:
🤦 missed that.

Reply:
(granted this is my first ts code, so could use some refinement)